instance/server/Router.elm

```diff
@@ -86,27 +87,34 @@ routePost ( req, res ) model =
                 |> runRouteWithErrorHandler
             )
     in
-        if url == routes.apply then
-            model
-                => generate generateSuccessPage
-        else if url == routes.signup then
-            model
-                => generate generateSignupPage
-        else if url == routes.startTest then
-            model
-                => generate generateTestPage
-        else if url == routes.login then
-            model
-                => generate generateAdminPage
-        else if url == routes.registerUser then
-            model
-                => generate generateSuccessfulRegistrationPage
-        else
-            model
-                => (handleError res (Task.fail "Route not found")
-                        |> runRouteWithErrorHandler
-                   )
-
+        case possibleRoute of
```

Comment:
This level of nesting feels unneeded. @etaque do you have a better way of structuring this? I would prefer it if we could use something like

```elm

type Route = Apply | Signup | NotFound String
```

where `NotFound` takes the path that the user tried to access